gebrueder-weiss-woocommerce.php

```diff
@@ -38,11 +39,39 @@
     };
 
     $plugin = GbWeiss::getInstance();
-    $authenticationClient = new OAuthAuthenticator(new GuzzleHttp\Client());
 
-    $authEndpoint = array_key_exists('GEBRUEDER_WEISS_OAUTH_URL', $_ENV) ? $_ENV['GEBRUEDER_WEISS_OAUTH_URL'] : "https://apitest.gw-world.com:443/authorize";
-    $authenticationClient->setAuthenticationEndpoint($authEndpoint);
+
+    $tokenEndpoint = env('GEBRUEDER_WEISS_OAUTH_TOKEN_URL', 'https://apitest.gw-world.com:443/token');
+
+    $authProvider = new GenericProvider([
+        // Has to be set as non-empty string to instantiate provider.
+        'clientId'                => 'clientId',
+        // Has to be set as non-empty string to instantiate provider.
+        'clientSecret'            => 'clientSecret',
+        'redirectUri'             => null,
+        'urlAuthorize'            => null,
+        'urlAccessToken'          => $tokenEndpoint,
+        'urlResourceOwnerDetails' => null
+    ]);
+    $authenticationClient = new OAuthAuthenticator($authProvider);
+
+
     $plugin->setAuthenticationClient($authenticationClient);
     $plugin->setOrderStateRepository(new OrderStateRepository());
     $plugin->initialize();
+    $plugin->updateToken();
```

Comment:
I would not execute update token here, since it will be exectued on every page load.

Reply:
ah yes, this was dev